src/Sylius/Bundle/CoreBundle/DependencyInjection/Compiler/BackwardsCompatibility/Symfony6PrivateServicesPass.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\DependencyInjection\Compiler\BackwardsCompatibility;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/** @internal */
+final class Symfony6PrivateServicesPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        // Some services are still fetched directly from the container in both Sylius and its extracted bundles (e.g. ResourceBundle)
+        // Before we fix it let's make them public to keep backward compatibility
+
+        $this->makeServicePublic('form.factory', $container);
+        $this->makeServicePublic('security.authorization_checker', $container);
+        $this->makeServicePublic('security.authorization_checker', $container);
+        $this->makeServicePublic('security.csrf.token_manager', $container);
+        $this->makeServicePublic('security.csrf.token_manager', $container);
+        $this->makeServicePublic('security.token_storage', $container);
+        $this->makeServicePublic('security.token_storage', $container);
+        $this->makeServicePublic('validator', $container);
+
+        if (str_starts_with((string) $container->getParameter('kernel.environment'), 'test')) {
+            $this->makeServicePublic('filesystem', $container);
+            $this->makeServicePublic('session.factory', $container);
+            $this->makeServicePublic('sylius.command_bus', $container);
+        }
```

Comment:
A slightly different topic - these are services used in our PHPUnit tests and, as far as I know, they _have_ to be taken directly from the container :/ That's why I make them public in a test (or test_cached) environment 🚀